django/thunderstore/community/models/community.py

```diff
@@ -206,3 +199,140 @@ def valid_review_statuses(self):
             PackageListingReviewStatus.approved,
             PackageListingReviewStatus.unreviewed,
         )
+
+
+@dataclass
+class AggregatedFields:
+    download_count: int = 0
+    package_count: int = 0
+
+
+class CommunityAggregatedFields(TimestampMixin, models.Model):
```

Comment:
## General comment regarding the implementation

The current implementation is a bit "everything happens at the same time"-esque. I'd suggest breaking it down to more atomic units / procedural phases e.g. as follows:

1. Bulk creation zeroed aggregates for communities that don't yet have them
2. Bulk updating existing aggregates

Ideally the task scope should be designed around potential resource constraints & parallelization. The above split works because both can be more or less done in the db directly, and if necessary in the future it could be split into a chunked iterator.

It would also be possible to replace the second task with one that handles the updating of an individual community if it's computationally expensive, as that's easy to parallelize by queuing each community update as a separate task. This also illustrates why you might want to bulk-create empty objects as a separate step entirely, or perhaps use e.g. Django signals to automatically create them on new object insertions.

To be clear, I'm not suggesting splitting the Celery task into two; just calling two different functions sequentially from it should suffice at least for now.

Reply:
Ready for another review methinks.